src/app/modules/group/pages/group-edit/group-edit.component.ts

```diff
@@ -74,28 +78,36 @@ export class GroupEditComponent implements OnDestroy, PendingChangesComponent {
     });
   }
 
-  getGroupChanges(): GroupChanges {
-    const description = this.groupForm.get('description')?.value as string;
-    const rootActivityId = this.groupForm.get('rootActivityId')?.value as string|null;
-    return {
-      name: this.groupForm.get('name')?.value as string,
-      description: description === '' ? null : description,
-      root_activity_id: rootActivityId,
-    };
-  }
-
   save(): void {
     if (!this.initialFormData) return;
 
     if (this.groupForm.invalid) {
       this.errorToast($localize`You need to solve all the errors displayed in the form to save changes.`);
       return;
     }
-
     this.groupForm.disable();
-    this.groupUpdateService.updateGroup(
-      this.initialFormData.id,
-      this.getGroupChanges(),
+
+    const rootActivity = this.groupForm.get('rootActivity')?.value as NoActivity|NewActivity|ExistingActivity;
+    const description = this.groupForm.get('description')?.value as string;
+
+    forkJoin({
+      id: of(this.initialFormData.id),
+      changes: forkJoin({
+        name: of(this.groupForm.get('name')?.value as string),
+        description: of(description === '' ? null : description),
+        root_activity_id: !isNewActivity(rootActivity) ? of(isExistingActivity(rootActivity) ? rootActivity.id : null) :
+          this.createItemService.create({
+            title: rootActivity.name,
+            type: rootActivity.itemType,
+            languageTag: 'en',
```

Comment:
add a " // FIXME " as it is not valid, it has to be fixed soon.

Reply:
fixed